src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -201,53 +212,55 @@ class UnixEvent
       m_state(initialState)
     {
         int st = pthread_mutex_init(&m_mutex, NULL);
-        ASSERT(st == NULL);
+        FATAL_ASSERT(st == 0, "Failed to initialize UnixEvent mutex");
```

Comment:
It would be nice to do a proper error handling for the initialization instead of error handling via FATAL_ASSERTs.